minions/async/src/main/java/io/github/microcks/minion/async/producer/KafkaProducerManager.java

```diff
@@ -225,6 +227,26 @@ public void publishMessage(String topic, String key, GenericRecord value, Set<He
       return null;
    }
 
+   /**
+    * Get the Kafka topic name corresponding to a AsyncMockDefinition, sanitizing all parameters.
+    */
+   public String getTopicName(AsyncMockDefinition definition, EventMessage eventMessage) {
+      // Produce service name part of topic name.
+      String serviceName = definition.getOwnerService().getName().replace(" ", "");
+      serviceName = serviceName.replace("-", "");
+      // Produce version name part of topic name.
+      String versionName = definition.getOwnerService().getVersion().replace(" ", "");
+      // Produce operation name part of topic name.
+      String operationName = definition.getOperation().getName();
+      if (operationName.startsWith("SUBSCRIBE ")) {
+         operationName = operationName.substring(operationName.indexOf(" ") + 1);
+      }
+      operationName = operationName.replace('/', '-');
+      operationName = ProducerManager.replacePartPlaceholders(eventMessage, operationName);
```

Comment:
The `io.github.microcks.util.URIBuilder` contains util methods for building URI from templates and parameters. There should be something reusable right there I think. Could you check please?

Reply:
See comment further below - I replied in the wrong place. I will switch to URIBuilder in another PR that I'll submit once this one is resolved. It involves moving a few things around.